### DISCUSSION TITLE
NFStream and DPDK

### DISCUSSION THREAD

**Discussion by elybes** [nfstream/nfstream#109] [Q&A]
Hello, can you help me solve this problem please ?
Does NFStream support **DPDK** ? because i would like to **capture packets directly from DPDK** interface and then parse them with nDPI NFStream. Is it possible to add plugins or functions to implement this solution ?
Thank you in advance.

**Comment by aouinizied** [comment]
Hi @elybes ,

Currently, NFStream engine is shipped (statically linked) with custom lipcap that enables AFPACKETv3 but not DPDK yet.
If you want to enable DPDK, this would be possible if you compile [**libpcap with DPDK support**][l1] and then compile the engine of nfstream using this libpcap version.
If you are willing to implement this, it will be a great contribution and will appreciate a PR. :)

[l1]: https://github.com/the-tcpdump-group/libpcap/blob/master/pcap-dpdk.c

**Reply by elybes** [reply]
Thank you for your answer !
So that means that i have to define a new libpcap function that supports DPDK and then import it to the engine ? or import it to another library ?... i mean where do i have to insert the new libpcap function ?

**Reply by aouinizied** [reply]
Once compiled with dpdk support, libpcap will abstract such function. What you need to do is first to compile libpcap with dpdk support and then integrate it as part of the engine.
Did you verify that you really need dpdk for your use case. I mean did you run tests on the traffic you need to monitor and observed drops?

**Reply by elybes** [reply]
Thank you for you answer ! I would like to explain the reason why i do need DPDK, so my project consists of capturing real time trafic packets using DPDK interface and then parse them and extract informations in order to create a machine learning algorithm that can identify applications type and classify them. So the DPDK interface is already set and built and I need to link it with NFStream so I can directly capture trafic with DPDK and parse it with NFStream because I found that NFStream fulfills all my needs for the project.

**Reply by drnpkr** [reply]
You should try reaching out to [Yellow-King21](https://github.com/nfstream/nfstream/issues?q=is%3Aissue+author%3AYellow-King21). To the best of my knowledge, he was recently working on exactly something like this. He might be able to give you some hints how to proceed with such implementation.

**Reply by elybes** [reply]
Thank you for your answer @drnpkr ! but sorry, it is not the same issue

**Reply by MalekChebli** [reply]
@aouinizied can you please explain more how to integrate dpdk as a part of the engine like you said here "What you need to do is first to compile libpcap with dpdk support and then integrate it as part of the engine."
Thank you in advance
